src/shogun/base/SGObject.cpp

```diff
@@ -1068,11 +1070,19 @@ CSGObject* CSGObject::get(const std::string& name)
 	if (auto* result = get_sgobject_type_dispatcher<CLabels>(name))
 		return result;
 
-
-	SG_ERROR(
-	    "Cannot get parameter %s::%s of type %s as object, not object type.\n",
-	    get_name(), name.c_str(),
-	    self->map[BaseTag(name)].get_value().type().c_str());
+	if (self->map.find(BaseTag(name)) != self->map.end())
+	{
+		SG_ERROR(
+		    "Cannot get parameter %s::%s of type %s as object, not object "
```

Comment:
Not sure I understand this error message, the second part

Reply:
ah maybe "Not a shogun object base type" or something like this?

Reply:
"Not a shogun CSGObject type"? Just to be more specific?

Reply:
Or does it not make sense here?

Reply:
i would just saw shogun as swig users might not know what sgobject is :)